coriolis/tests/api/v1/test_minion_pools.py

```diff
@@ -0,0 +1,431 @@
+                exception_raised,
+                self.minion_pools._validate_updated_environment_options,
+                context,
+                minion_pool,
+                enviroment_options
+            )
+        else:
+            self.minion_pools._validate_updated_environment_options(
+                context,
+                minion_pool,
+                enviroment_options
+            )
+
+        if platform == constants.PROVIDER_PLATFORM_SOURCE:
+            (mock_validate_endpoint_source_minion_pool_options.
+                assert_called_once_with)(
+                    context, endpoint_id, enviroment_options)
+        if platform == constants.PROVIDER_PLATFORM_DESTINATION:
+            (mock_validate_endpoint_destination_minion_pool_options.
+                assert_called_once_with)(
+                    context, endpoint_id, enviroment_options)
+
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_validate_updated_environment_options')
+    @mock.patch.object(api.API, 'get_minion_pool')
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_check_pool_retention_strategy')
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_check_pool_numeric_values')
+    @ddt.file_data('data/minion_pools_validate_update_body.yml')
+    def test__validate_update_body(
+        self,
+        mock_check_pool_numeric_values,
+        mock_check_pool_retention_strategy,
+        mock_get_minion_pool,
+        mock_validate_updated_environment_options,
+        config,
+        exception_raised,
+    ):
+        body = config["body"]
+        minion_pool = body["minion_pool"]
+        environment_options = minion_pool.get('environment_options', {})
+        minion_retention_strategy = minion_pool.get(
+            'minion_retention_strategy', "")
+        validate_get_minion_pool = config.get("validate_get_minion_pool", None)
+        minimum_minions = minion_pool.get('minimum_minions', 1)
+        maximum_minions = minion_pool.get('maximum_minions', 1)
+        minion_max_idle_time = minion_pool.get('minion_max_idle_time', 1)
+        mock_context = mock.Mock()
+        id = mock.sentinel.id
+        mock_get_minion_pool.return_value = {
+            'minimum_minions': 1,
+            'maximum_minions': 1
+        }
+
+        if exception_raised:
+            self.assertRaisesRegex(
+                Exception,
+                exception_raised,
+                testutils.get_wrapped_function(
+                    self.minion_pools._validate_update_body),
+                self.minion_pools,
+                id,
+                mock_context,
+                body
+            )
+        else:
+            testutils.get_wrapped_function(
+                self.minion_pools._validate_update_body)(
+                    self.minion_pools,
+                    id,
+                    mock_context,
+                    body,
+            )
+
```

Comment:
Please also add an `expected_result` to the yml file to assert the final result of the called method, and to also check this properly:
https://github.com/cloudbase/coriolis/blob/dadad51ee6bd7576e4c8b96d215b71e3e85fe337/coriolis/api/v1/minion_pools.py#L173-L176

Ideally, you could also add additional key-values to the body, which `expected_result` should not contain.